source/backends/ubuntu/ubupkg.d

```diff
@@ -0,0 +1,167 @@
+                return getTranslations (langpackdomain, text);
+            }
+        } catch {
+            return null;
+        }
+    }
+
+private:
+    string globalTmpDir;
+    string langpackDir;
+    string localeDir;
+    string[] langpackLocales;
+    Array!Package allPackages;
+
+    void extractLangpacks ()
+    {
+        import std.algorithm : filter, map;
+        import std.array : appender, array, split;
+        import std.file : dirEntries, exists, SpanMode, readText;
+        import std.parallelism : parallel;
+        import std.path : baseName;
+        import std.process : Pid, spawnProcess, wait;
+        import std.string : splitLines, startsWith;
+
+        auto path = buildPath (langpackDir, "usr", "share", "locale-langpack");
+
+        if (!langpackDir.exists) {
+            foreach (pkg; allPackages) {
+                if (!pkg.name.startsWith ("language-pack"))
+                    continue;
+
+                /* :( */
+                UbuntuPackage upkg = cast (UbuntuPackage) pkg;
+
+                logDebug ("Extracting %s", pkg.name);
+                upkg.extractPackage (langpackDir);
+            }
+
+            auto supportedd = buildPath (langpackDir, "var", "lib", "locales", "supported.d");
+
+            mkdirRecurse (localeDir);
+
+            foreach (locale; parallel (supportedd.dirEntries (SpanMode.shallow), 5))
+            {
+                    foreach (ref line; locale.readText.splitLines) {
+                            auto components = line.split (" ");
+                            auto localecharset = components[0].split (".");
+
+                            auto outdir = buildPath (localeDir, components[0]);
+                            logDebug ("Generating locale in %s", outdir);
+
+                            auto pid = spawnProcess (["/usr/bin/localedef",
+                                                      "--no-archive",
+                                                      "-i",
+                                                      localecharset[0],
+                                                      "-c",
+                                                      "-f",
+                                                      components[1],
+                                                      outdir]);
+
+                            scope (exit) wait (pid);
+                    }
+            }
+
+        }
+
+        if (langpackLocales is null)
+            langpackLocales = localeDir.dirEntries (SpanMode.shallow)
+                .filter!(f => f.isDir)
+                .map!(f => f.name.baseName)
+                .array;
+    }
+
+    string[string] getTranslations (const string domain, const string text)
+    {
+        import core.stdc.locale;
+
+        import std.c.stdlib : getenv, setenv, unsetenv;
+        import std.string : fromStringz, toStringz;
+
+        foreach (ref var; ["LC_ALL", "LANG", "LANGUAGE", "LC_MESSAGES"])
+                unsetenv (var.toStringz);
```

Comment:
Hmm, not sure... If we - for whatever reason - add translation stuff to asgen in future, we wouldn't need to chase down bugs which for some reason only occur with the Ubuntu backend.
I am a bit worried that this might have an impact on other parts of asgen, mainly the stuff handling fonts / drawing stuff (you never know...). So maybe unsetting the env vars again is a good idea, so we don't have any of the external libraries pick up some strange locale and do weird things because of that.